homeassistant/components/neato/__init__.py

```diff
@@ -79,7 +79,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     session = config_entry_oauth2_flow.OAuth2Session(hass, entry, implementation)
 
-    neato_session = api.ConfigEntryAuth(hass, entry, session)
+    neato_session = api.ConfigEntryAuth(hass, entry, session.implementation)
```

Comment:
The `implementation` variable is already there.

```suggestion
    neato_session = api.ConfigEntryAuth(hass, entry, implementation)
```

Reply:
Doesn't work ;-)

(and would make session totally redundant as is never access)

Reply:
Then remove the session. Then, I think we don't need that here since the session is also created by `ConfigEntryAuth`

Reply:
ha, I didn't see this linked PR, was just responding in Discord, but sounds like @Santobert and I are on the same page (see comment in linked issue)